### PR TITLE
Fix app container build, fix security vulnerabilities in dependencies

### PR DESCRIPTION
**Pull request by BjoernAtBosch**
## Checklist - Manual tasks

* [x] Examples are executing successfully
* [ ] Created/updated unit tests. Code Coverage percentage on new code shall be >= 80%.
* [ ] Created/updated integration tests.
* [x] Devcontainer can be opened successfully
* [x] Devcontainer can be opened successfully behind a corporate proxy
* [x] Devcontainer can be re-built successfully
* [ ] Extended the documentation (e.g. README.md, CONTRIBUTING.md, Velocitas)

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -80,7 +78,24 @@ def imports(self):
...
+        # Workaround: Pin recipe revision for transient dependency googleapis for enabling the container build
+        self.requires("googleapis/cci.20221108@#e4bebdfa02f3b6f93bae1d5001b8d439")
+        # Workaround: 
+        self.requires("paho-mqtt-c/1.3.9@#0421671a9f4e8ccfa5fc678cfb160394")
+
     def build_requirements(self):
         # 'build' context (protoc.exe will be available)
-        self.tool_requires("protobuf/3.21.4")
+        self.tool_requires("protobuf/3.21.9")
         self.tool_requires("grpc/1.48.0")
```
Comment (kse3hi) [inline conanfile.py:100]: grpc 1.48.0 is required 3.21.4, therefore we need to update it on 1.50.1

File: conanfile.py [hunk 2]
```diff
@@ -80,7 +78,24 @@ def imports(self):
         self.copy("license*", src=".", dst="./licenses",
                   folder=True, ignore_case=True)
 
+    def requirements(self):
+        # Direct dependencies:
+        self.requires("fmt/9.1.0")
+        self.requires("nlohmann_json/3.11.2")
+        self.requires("grpc/1.48.0")
```
Comment (kse3hi) [inline conanfile.py:85]: grpc 1.48.0 is required 3.21.4, therefore we need to update it on 1.50.1

File: conanfile.py [hunk 3]
```diff
@@ -23,8 +23,6 @@ class VehicleAppCppSdkConan(ConanFile):
     license = "Apache-2.0"
     url = "https://github.com/eclipse-velocitas/vehicle-app-cpp-sdk"
     description = "The Vehicle App SDK for c++ allows to create Vehicle Apps from the Velocitas development model in the c++ programming language."
-    # Workaround: Pin recipe revision for transient dependency googleapis for enabling the container build
-    requires = "openssl/1.1.1q", "libcurl/7.84.0", "nlohmann_json/3.11.2", "paho-mqtt-cpp/1.2.0", "grpc/1.48.0", "protobuf/3.21.4", "cpr/1.9.0", "fmt/9.1.0", "zlib/1.2.13", "googleapis/cci.20221108@#e4bebdfa02f3b6f93bae1d5001b8d439"
```
Comment (kse3hi) [inline conanfile.py:27]: You've removed libcurl and zlib, was it intentional?

File: conanfile.py [hunk 4]
```diff
@@ -80,7 +78,24 @@ def imports(self):
...
+        self.requires("grpc/1.48.0")
+        self.requires("protobuf/3.21.9")
+        self.requires("paho-mqtt-cpp/1.2.0")
+        self.requires("cpr/1.9.3")
+
+        # Transient dependencies:
+        self.requires("openssl/1.1.1t")
+        # Workaround: Pin recipe revision for transient dependency googleapis for enabling the container build
+        self.requires("googleapis/cci.20221108@#e4bebdfa02f3b6f93bae1d5001b8d439")
+        # Workaround:
```
Comment (kse3hi) [inline conanfile.py:94]: Please add more details in the description comment.
Please provide a ticket when the proper solution is going to be provided.

**Review by kse3hi — CHANGES_REQUESTED** [review]

File: conanfile.py (hunk 3 above)
Comment (kse3hi) [reply conanfile.py:27]: Which version of libcurl is installed if it was not explicitly mentioned?

**Comment by kse3hi** [comment]
General: please follow the pattern of branch naming:
Feature: feature/feature_name
Bug: bugfix/bugfix_name
User specific: user/user_id

**Comment by kse3hi** [comment]
Please don't forget to add "--udpate" flag into ./install_dependencies.sh in line 119

File: conanfile.py (hunk 1 above)
Comment (BjoernAtBosch) [reply conanfile.py:87]: This depends on the revision of the Conan recipe. Basically you are right, the latest recipe revision of grpc/1.48.0 refers to protobuf/3.21.4.
But: Conan "isn't the world" and you could decide to use another package manager or plugging library together manually. Means, due to semantic versioning the increase to 3.21.9 is just an increase of the patch (or micro) version of protobuf, so I definitively would expect downward compatible behaviour of it. (Another talk would be if a grpc would require 3.21.9 and we would use 3.21.4. In that situation the grpc lib could expect some certain behaviour available from 3.21.9 onwards, which is maybe not available in 3.21.4. Nevertheless, you'll find differing recipe revisions for grpc/1.50.1 being happy with protobuf 3.21.4 and 3.21.9 ...)

File: conanfile.py (hunk 3 above)
Comment (BjoernAtBosch) [reply conanfile.py:27]: The drawn libcurl version is now 7.87.0.

File: conanfile.py (hunk 4 above)
Comment (BjoernAtBosch) [reply conanfile.py]: I would like to add a ticket, but the problem is that we currently support bug issues and security issues in GitHub and I don't like to put links to our internal tooling here.

**Comment by BjoernAtBosch** [comment]
I wouldn't like to add the --update flag, as I'm not sure about its general behaviour. Currently the issue occurs in local environment only, where the wrong revision could be cached. This should not happen in CI workflows.

File: conanfile.py (hunk 1 above)
Comment (BjoernAtBosch) [reply conanfile.py]: grpc 1.50.1 seems to work, will test tomorrow

**Review by kse3hi — APPROVED** [review]
LGTM

**Pull request merged**